chain/db.go

```diff
@@ -181,6 +198,135 @@ func NewMemDB() *MemDB {
 	}
 }
 
+type cacheBucket struct {
+	mb memBucket
+	db DBBucket
+}
+
+func (b cacheBucket) Get(key []byte) []byte {
+	if val := b.mb.Get(key); val != nil {
+		return val
+	} else if _, ok := b.mb.db.dels[b.mb.name][string(key)]; ok {
+		return nil
+	}
+	return b.db.Get(key)
+}
+
+func (b cacheBucket) Put(key, value []byte) error {
+	err := b.mb.Put(key, value)
+	return err
+}
+
+func (b cacheBucket) Delete(key []byte) error {
+	return b.mb.Delete(key)
+}
+
+func (b cacheBucket) Iter() iter.Seq2[[]byte, []byte] {
+	return b.db.Iter() // NOTE: imperfect
```

Comment:
😬, maybe add disclaimer to `NewCacheDB`?

Reply:
I'll just implement it properly lol